packages/regex/match_iterator.pony

```diff
@@ -0,0 +1,39 @@
+class MatchIterator is Iterator[Match]
+  """
+  MatchIterator allows for calling code to repeatedly perform the same match
+  against a subject string as an iterator. This lets callers repeat the match 
+  until no more matches exist.
+  """
+  let _regex: Regex box
+  let _subject: String 
+  var _offset: USize = 0
+
+  new create(regex': Regex box, subject': String, offset': USize = 0) =>   
+    """
+    Creates a new Match Iterator from a regular expression and a subject 
+    string 
+    """ 
+    _regex = regex'
+    _subject = consume subject'
+    _offset = offset'
+
+  fun has_next() : Bool =>
+    """
+    Indicates whether there is another match available
+    """
+    try
+      let m = _regex(_subject, _offset)?
+      true
+    else
+      false
+    end 
+
+  fun ref next() : Match? =>
```

Comment:
Style-wise, the `:` here should not have a preceding space, and the `?` should.

When in doubt on style, check the style guide as well as surrounding code.